applications/StructuralMechanicsApplication/custom_elements/small_displacement_shifted_boundary_element.h

```diff
@@ -0,0 +1,314 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+
+// Application includes
+#include "small_displacement.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template<std::size_t TDim>
+class SmallDisplacementShiftedBoundaryElement : public SmallDisplacement
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of SmallDisplacementShiftedBoundaryElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(SmallDisplacementShiftedBoundaryElement);
+
+    typedef SmallDisplacement BaseType;
+
+    static constexpr std::size_t NumNodes = TDim + 1;
+
+    static constexpr std::size_t StrainSize = 3 * (TDim-1);
+
+    static constexpr std::size_t BlockSize = TDim;
+
+    static constexpr std::size_t LocalSize = NumNodes * TDim;
```

Comment:
Okay you assume simplex (triangle +  tetrahedra)

Reply:
Yes, at least for the moment since our main levelset calculation routines assume this.

Reply:
perhaps add some compiletime checks for the geometries it is used with? Or at least runtime